compiler/src/iree/compiler/Dialect/Flow/Transforms/SetEncoding.cpp

```diff
@@ -86,24 +88,32 @@ static Value setEncoding(OpBuilder &builder, Location loc, Value source,
 static LinalgExt::EncodingAttr makeEncoding(OpBuilder &builder,
                                             LinalgExt::EncodingUser user,
                                             LinalgExt::EncodingRole role,
+                                            TypeRange operandTypes,
                                             Type originalType) {
   auto *context = builder.getContext();
   auto userAttr = LinalgExt::EncodingUserAttr::get(context, user);
   auto roleAttr = LinalgExt::EncodingRoleAttr::get(context, role);
+  SmallVector<Attribute> elemTypeAttrs;
+  for (auto t : operandTypes) {
+    elemTypeAttrs.emplace_back(
+        TypeAttr::get(t.cast<ShapedType>().getElementType()));
+  }
```

Comment:
[optional] we can use `llvm::map_to_vector` here.

```suggestion
  SmallVector<Attribute> elemTypeAttrs =
      llvm::map_to_vector(operandTypes, [](auto t) {
        return TypeAttr::get(t.cast<ShapedType>().getElementType());
      });
```